src/software/firmware/srcs/telemetry.cpp

```diff
@@ -0,0 +1,479 @@
+    Serial6.write(MODE);
+    Serial6.print("\t");
+    Serial6.write(min8);
+    Serial6.print("\t");
+    Serial6.write(max8);
+    Serial6.print("\t");
+
+    byte min32Value[4];  // 32 bits
+    min32Value[0] = (min32 >> 24) & FIRST_BYTE;
+    min32Value[1] = (min32 >> 16) & FIRST_BYTE;
+    min32Value[2] = (min32 >> 8) & FIRST_BYTE;
+    min32Value[3] = min32 & FIRST_BYTE;
+    Serial6.write(min32Value, 4);
+
+    Serial6.print("\t");
+
+    byte max32Value[4];  // 32 bits
+    max32Value[0] = (max32 >> 24) & FIRST_BYTE;
+    max32Value[1] = (max32 >> 16) & FIRST_BYTE;
+    max32Value[2] = (max32 >> 8) & FIRST_BYTE;
+    max32Value[3] = max32 & FIRST_BYTE;
+    Serial6.write(max32Value, 4);
+
+    Serial6.print("\n");
+#endif
+}
+
+// cppcheck-suppress unusedFunction
+void sendDataSnapshot(uint16_t centileValue,
+                      uint16_t pressureValue,
+                      CyclePhases phase,
+                      CycleSubPhases subPhase,
+                      uint8_t blowerValvePosition,
+                      uint8_t patientValvePosition,
+                      uint8_t blowerRpm,
+                      uint8_t batteryLevel) {
+#if HARDWARE_VERSION == 1
+    (void)centileValue;
+    (void)pressureValue;
+    (void)phase;
+    (void)subPhase;
+    (void)blowerValvePosition;
+    (void)patientValvePosition;
+    (void)blowerRpm;
+    (void)batteryLevel;
+#elif HARDWARE_VERSION == 2
+    uint8_t phaseValue;
+    if ((phase == CyclePhases::INHALATION) && (subPhase == CycleSubPhases::INSPIRATION)) {
+        phaseValue = 17u;  // 00010001
+    } else if ((phase == CyclePhases::INHALATION)
+               && (subPhase == CycleSubPhases::HOLD_INSPIRATION)) {
+        phaseValue = 18u;  // 00010010
+    } else if ((phase == CyclePhases::EXHALATION) && (subPhase == CycleSubPhases::EXHALE)) {
+        phaseValue = 68u;  // 01000100
+    } else {
+        phaseValue = 0u;
+    }
+
+    Serial6.write("D:");
+    Serial6.write((uint8_t)1u);
+
+    Serial6.write(static_cast<uint8_t>(strlen(VERSION)));
+    Serial6.print(VERSION);
+
+    byte deviceId1[4];  // 32 bits
+    deviceId1[0] = (LL_GetUID_Word0() >> 24) & FIRST_BYTE;
+    deviceId1[1] = (LL_GetUID_Word0() >> 16) & FIRST_BYTE;
+    deviceId1[2] = (LL_GetUID_Word0() >> 8) & FIRST_BYTE;
+    deviceId1[3] = LL_GetUID_Word0() & FIRST_BYTE;
+    Serial6.write(deviceId1, 4);
+
+    byte deviceId2[4];  // 32 bits
+    deviceId2[0] = (LL_GetUID_Word1() >> 24) & FIRST_BYTE;
+    deviceId2[1] = (LL_GetUID_Word1() >> 16) & FIRST_BYTE;
+    deviceId2[2] = (LL_GetUID_Word1() >> 8) & FIRST_BYTE;
+    deviceId2[3] = LL_GetUID_Word1() & FIRST_BYTE;
+    Serial6.write(deviceId2, 4);
+
+    byte deviceId3[4];  // 32 bits
+    deviceId3[0] = (LL_GetUID_Word2() >> 24) & FIRST_BYTE;
+    deviceId3[1] = (LL_GetUID_Word2() >> 16) & FIRST_BYTE;
+    deviceId3[2] = (LL_GetUID_Word2() >> 8) & FIRST_BYTE;
+    deviceId3[3] = LL_GetUID_Word2() & FIRST_BYTE;
+    Serial6.write(deviceId3, 4);
```

Comment:
This part is constant so I think it should be generated only once.

Reply:
Done!

Reply:
@dsferruzza 
My concern was to read and store each deviceIdx once then each time you call the telemetry, simply reuse those stored deviceIdx as they are constant.

Reply:
Done! That's much better like this! Sorry I was slow on this one 😴

Reply:
No worries ;)